src/Components/Atbash/AtbashInput.js

```diff
@@ -19,45 +42,115 @@ class Atbash extends Component {
     return letters;
   };
 
+  putCharsInSpans = (string) => {
+    let map = Array.prototype.map;
+    let spans = map.call(string, char => {
+        if (char === " ") {
+            return <span>&nbsp;</span>
+        } else {
+            return <span>{char}</span>;
+        }
+    });
+    return spans;
+  }
+
+
+  decryptHelper = (i, index) => {
+    let original = this.state.messages[i]
+    original = atbashEncode( original.substring(0, index+1)) + original.substring(index + 1);
+    //original = original.substring(0, index) + atbashEncode(original[index]) + original.substring(index + 1);
+    let newmsgs = Array.from(this.state.messages)
+    newmsgs.splice(i, 1, original)
+    console.log(newmsgs[i])
+    this.setState({
+      messages: newmsgs
+    });
+  }
```

Comment:
This can be boiled down significantly:

```suggestion
  decryptHelper = (msgIdx, substrIdx) => {
    let original = this.state.messages[msgIdx];
    original = atbashEncode( original.substring(0, substrIdx+1)) + original.substring(substrIdx + 1);
    this.setState({
      messages: Array.from(this.state.messages).splice(msgIdx, 1, original),
    });
  }
```